terraform/ec2/main.tf

```diff
@@ -260,6 +264,38 @@ resource "null_resource" "start_collector" {
   }
 }
 
+resource "aws_ssm_parameter" "setup_aoc_config" {
+  count = var.ssm_mode ? 1 : 0
+  name  = format("aoc-config-%s", module.common.testing_id)
+  type  = "String"
+  value = var.ssm_config
+}
+
+resource "null_resource" "install_collector_from_ssm" {
+  depends_on = [null_resource.check_patch]
+  count = var.ssm_mode ? 1 : 0
+
+  provisioner "remote-exec" {
+    inline = [
+      local.ami_family["wait_cloud_init"],
+    ]
+
+    connection {
+      type = local.connection_type
+      user = local.login_user
+      private_key = local.connection_type == "ssh" ? local.private_key_content : null
+      password = local.connection_type == "winrm" ? rsadecrypt(aws_instance.aoc.password_data, local.private_key_content) : null
+      host = aws_instance.aoc.public_ip
+    }
+  }
+
+    provisioner "local-exec" {
+    command = <<-EOT
```

Comment:
There should be indent here, if you install terraform plugin, it can format the file like other languages. Though existing format in ec2/main.tf is not consistent, and you can configure tf to align properties in different ways.

I will create another pr for the formatting, format the entire file will the pr hard to read.

![Screen Shot 2021-04-03 at 2 45 39 PM](https://user-images.githubusercontent.com/72180622/113492308-61605400-948b-11eb-8102-d4e4d43e3684.png)

Reply:
Agree